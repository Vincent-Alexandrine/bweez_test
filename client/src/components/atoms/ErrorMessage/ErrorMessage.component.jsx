import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import Styled from './ErrorMessage.styled.js';

function ErrorMessage({ err }) {
  const [t] = useTranslation();
  return (
    <Styled>
      {t(`error.${err}`)}
    </Styled>
  );
}

ErrorMessage.propTypes = {
  err: PropTypes.number,
};

export default ErrorMessage;
