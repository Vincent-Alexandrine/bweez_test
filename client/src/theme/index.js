import ThemeService from './theme.service.js';

export default ThemeService;
