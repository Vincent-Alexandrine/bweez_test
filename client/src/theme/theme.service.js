/*
  getTheme(string) -> theme
  addTheme(string, theme) -> void || true || promise
  removeTheme(string) -> void || true || promise

  validateTheme(theme) -> boolean
*/

const themesMap = new Map();

class ThemeService {
  static instance;

  constructor() {
    if (ThemeService.instance)
      return ThemeService.instance;

    ThemeService.instance = this;
  }

  get(name) {
    return themesMap.get(name);
  }

  add({ name, ...theme }) {
    themesMap.set(name, theme);
  }

  names() {
    return [...themesMap.keys()];
  }

  remove(name) {
    return themesMap.delete(name);
  }
}

const singleton = new ThemeService();

export default singleton;
