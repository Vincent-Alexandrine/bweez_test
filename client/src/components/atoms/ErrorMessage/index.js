import ErrorMessage from './ErrorMessage.component.jsx';

export default ErrorMessage;
