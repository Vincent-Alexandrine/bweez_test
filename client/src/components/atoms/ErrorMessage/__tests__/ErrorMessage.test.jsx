import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import theme from 'Styles/vManuscript';

import ErrorMessage from '../ErrorMessage.component.jsx';
import Styled from '../ErrorMessage.styled.js';

test('ErrorMessage component should match snapshot', () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ErrorMessage />
      </ThemeProvider>
    );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('ErrorMessage styled component should match snapshot', () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Styled />
      </ThemeProvider>
    );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
