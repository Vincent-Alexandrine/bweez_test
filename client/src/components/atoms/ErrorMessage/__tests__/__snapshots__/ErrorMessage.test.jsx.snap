// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component should match snapshot 1`] = `
<div
  className=""
>
  error.undefined
</div>
`;

exports[`ErrorMessage styled component should match snapshot 1`] = `
<div
  className=""
/>
`;
